index.js

```diff
@@ -40,8 +40,11 @@ class ExecutorRouter extends Executor {
             const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
             const executorPlugin = new ExecutorPlugin(options);
 
-            // Set the default executor
-            if (!this.defaultExecutor && !config.defaultPlugin) {
+            if (defaultPlugin && defaultPlugin === plugin.name) {
```

Comment:
It's been there https://github.com/screwdriver-cd/executor-router/pull/7/files#diff-168726dbe96b3ce427e7fedce31bb0bcR54